ldms/src/sampler/clock/Makefile.am

```diff
@@ -0,0 +1,16 @@
+pkglib_LTLIBRARIES =
+lib_LTLIBRARIES =
+dist_man7_MANS =
+dist_man1_MANS =
+
+AM_CPPFLAGS = @OVIS_INCLUDE_ABS@
+AM_LDFLAGS = @OVIS_LIB_ABS@
+COMMON_LIBADD = -lsampler_base -lldms -lovis_util -lcoll \
+		@LDFLAGS_GETTIME@
+
+if ENABLE_CLOCK
```

Comment:
These conditionals are all redundant. They are already taken care of in the in conditional SUBDIRS in the Makefile.am on level up.

Reply:
I will remove them from the level up then, since the details controlled should be near the control statement for maintainability and for simplicity of clone-modify development.

Reply:
I think staying with the conditionals around SUBDIRS is actually the way to go right now.

There is an autotools bug that would require extra handling of the man page if you move to only doing the conditional the per-sampler automake files. And we clearly are mostly using the SUBDIRS approach these days. I'm not sure that I see any different in difficult for "clone-modify development" either.

Reply:
Can you cite the automake bug you would have us code around? I've been testing on rhel7 (the oldest thing we support) and haven't managed to hit any problems with distcheck. I did notice inconsistent use elsewhere of extra_dist for a man page.

Reply:
@tom95858 can you clarify which subdirs approach you prefer, a or b?
A)
```
if xyz_enabled
subdirs += xyz
endif
```

or

B)
```
subdirs += xyz
# xyz/Makefile.am from the plugin author handles the conditionals
subdirs += pdq
subdir += sl
```

Reply:
> Can you cite the automake bug you would have us code around? I've been testing on rhel7 (the oldest thing we support) and haven't managed to hit any problems with distcheck. I did notice inconsistent use elsewhere of extra_dist for a man page.

Yes, there needed to be an EXTRA_DIST of the manpage outside of the conditional when the man page line is inside the conditional. distcheck wouldn't necessarily find the issue. The testing probably needs to go like this:

1. "configure" ldms to explicitly disable a plugin that uses this idion
2. generate a dist tarball with "make dist"
3. Check if tarball contains the man page in question

Reply:
@morrone and to be clear, the (A) approach of my question to Tom prevents this bug being triggered?

Reply:
I think that is the case.